snapcraft/project/_project.py

```diff
@@ -47,10 +52,11 @@ def __init__(
         else:
             self.info = ProjectInfo(snapcraft_yaml_file_path=snapcraft_yaml_file_path)
 
-        # These paths maintain backwards compatibility.
-        self._internal_dir = os.path.join(self.project_dir, "snap", ".snapcraft")
-        self._global_state_file = os.path.join(self._internal_dir, "state")
+        self._global_state_file = os.path.join(internal_dir, "state")
+        self.project_dir = project_dir
```

Comment:
Do plugins need access to this? Perhaps `self._project_dir` would be better.

Reply:
Plugins do not. Just our internal stuff, making it "protected" is a good call, we can other modules to use it, but only the ones we control.
It is somewhat interesting though as plugins already use private packages, python really doesn't provide the framework for this to be nicely designed.

Reply:
removed though, not needed at all

Reply:
> python really doesn't provide the framework for this to be nicely designed.

Yeah I've been feeling that way as well :cry: .

Reply:
issue taken care of